From 8393b61ba876c8e1614275c97767f9b06b889f48 Mon Sep 17 00:00:00 2001
From: Hans Wennborg <hans@chromium.org>
Date: Mon, 30 Jun 2025 07:45:33 -0700
Subject: [PATCH] [rust] Remove __rust_no_alloc_shim_is_unstable

We rely on __rust_no_alloc_shim_is_unstable_v2 now, after having rolled
past https://github.com/rust-lang/rust/pull/141061 in
https://crrev.com/1477839

Bug: 422538133
Change-Id: Ia5be23a0231b82d9f090b706a2fcc64f58b8f331
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/6686150
Auto-Submit: Hans Wennborg <hans@chromium.org>
Commit-Queue: Łukasz Anforowicz <lukasza@chromium.org>
Reviewed-by: Łukasz Anforowicz <lukasza@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1480510}
---

diff --git a/build/rust/allocator/lib.rs b/build/rust/allocator/lib.rs
index 142c6e60..29b3af1 100644
--- a/build/rust/allocator/lib.rs
+++ b/build/rust/allocator/lib.rs
@@ -90,12 +90,6 @@
     #[linkage = "weak"]
     fn __rust_no_alloc_shim_is_unstable_v2() {}
 
-    // TODO(crbug.com/422538133) Remove after rolling past
-    // https://github.com/rust-lang/rust/pull/141061
-    #[no_mangle]
-    #[linkage = "weak"]
-    static __rust_no_alloc_shim_is_unstable: u8 = 0;
-
     // Mangle the symbol name as rustc expects.
     #[rustc_std_internal_symbol]
     #[allow(non_upper_case_globals)]
